x/ecocredit/client/tx.go

```diff
@@ -52,14 +57,139 @@ func txCreateClass() *cobra.Command {
 				return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "base64_metadata is malformed, proper base64 string is required")
 			}
 
-			c, ctx, err := mkMsgClient(cmd)
+			c, err := newMsgSrvClient(cmd)
 			if err != nil {
 				return err
 			}
-			_, err = c.CreateClass(cmd.Context(), &ecocredit.MsgCreateClassRequest{
+			msg := ecocredit.MsgCreateClassRequest{
 				Designer: args[0], Issuers: issuers, Metadata: b,
-			})
-			return mkTx(ctx, cmd, err)
+			}
+			_, err = c.client.CreateClass(cmd.Context(), &msg)
+			return c.send(err)
+		},
+	}
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
+}
+
+func txCreateBatch() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "create_batch [issuer] [class_id] [metadata] [issuance]",
+		Short: `Issues new credit batach.
+Parameters:
+  issuer:    issuer address
+  class_id:  credit class
+  metadata:  base64 encoded issuance metadata
+  issuance:  JSON encode issuance list,
+             eg: '[{"recipient": "a1", "tradeableUnits": "10", "retiredUnits": "2"}]'`,
```

Comment:
Feels like we could have a cleaner interface for this. Maybe something to explore after the collabathon.

For one- we may want to handle not setting tradableUnits or retiredUnits to be set to a default value of 0 there's no json value provided.

Reply:
This is what happens. If the value is not set, it will go to default (0, "", nil...)

Reply:
BTW, I was thinking to use YAML instead of JSON. What do you think?